File: Unigram/Unigram.Tasks/TelegramGameProxy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.Foundation.Metadata;

namespace Unigram.Tasks
{
    public delegate void TelegramGameProxyDelegate(bool withMyScore);

    [AllowForWeb]
    public sealed class TelegramGameProxy
    {
        private TelegramGameProxyDelegate _callback;

        public TelegramGameProxy(TelegramGameProxyDelegate callback)
        {
            _callback = callback;
        }

        public void PostEvent(string eventName, string eventData)
        {
            if (eventName.Equals("share_game"))
            {
                _callback.Invoke(false);
            }
            else if (eventName.Equals("share_score"))
            {
                _callback.Invoke(true);
            }
        }
    }
}

File: Unigram/Unigram.Native/Proxies/TelegramGameProxy.cpp
#include "pch.h"
#include "Helpers\LibraryHelper.h"
#include "TelegramGameProxy.h"

using namespace Unigram::Native;
using namespace Platform;

TelegramGameProxy::TelegramGameProxy(TelegramGameProxyDelegate^ delegate)
{
	_delegate = delegate;
}

void TelegramGameProxy::PostEvent(String^ eventName, String^ eventData)
{
	if (eventName->Equals(L"share_game"))
	{
		_delegate(false);
	}
	else if (eventName->Equals("share_score"))
	{
		_delegate(true);
	}
}